bundles/org.openhab.binding.pihole/src/main/java/org/openhab/binding/pihole/internal/PiHoleHandler.java

```diff
@@ -0,0 +1,292 @@
+            return;
+        }
+
+        updateDecimalState(DOMAINS_BEING_BLOCKED_CHANNEL, localDnsStatistics.getDomainsBeingBlocked());
+        updateDecimalState(DNS_QUERIES_TODAY_CHANNEL, localDnsStatistics.getDnsQueriesToday());
+        updateDecimalState(ADS_BLOCKED_TODAY_CHANNEL, localDnsStatistics.getAdsBlockedToday());
+        updateDecimalState(UNIQUE_DOMAINS_CHANNEL, localDnsStatistics.getUniqueDomains());
+        updateDecimalState(QUERIES_FORWARDED_CHANNEL, localDnsStatistics.getQueriesForwarded());
+        updateDecimalState(QUERIES_CACHED_CHANNEL, localDnsStatistics.getQueriesCached());
+        updateDecimalState(CLIENTS_EVER_SEEN_CHANNEL, localDnsStatistics.getClientsEverSeen());
+        updateDecimalState(UNIQUE_CLIENTS_CHANNEL, localDnsStatistics.getUniqueClients());
+        updateDecimalState(DNS_QUERIES_ALL_TYPES_CHANNEL, localDnsStatistics.getDnsQueriesAllTypes());
+        updateDecimalState(REPLY_UNKNOWN_CHANNEL, localDnsStatistics.getReplyUnknown());
+        updateDecimalState(REPLY_NODATA_CHANNEL, localDnsStatistics.getReplyNoData());
+        updateDecimalState(REPLY_NXDOMAIN_CHANNEL, localDnsStatistics.getReplyNXDomain());
+        updateDecimalState(REPLY_CNAME_CHANNEL, localDnsStatistics.getReplyCName());
+        updateDecimalState(REPLY_IP_CHANNEL, localDnsStatistics.getReplyIP());
+        updateDecimalState(REPLY_DOMAIN_CHANNEL, localDnsStatistics.getReplyDomain());
+        updateDecimalState(REPLY_RRNAME_CHANNEL, localDnsStatistics.getReplyRRName());
+        updateDecimalState(REPLY_SERVFAIL_CHANNEL, localDnsStatistics.getReplyServFail());
+        updateDecimalState(REPLY_REFUSED_CHANNEL, localDnsStatistics.getReplyRefused());
+        updateDecimalState(REPLY_NOTIMP_CHANNEL, localDnsStatistics.getReplyNotImp());
+        updateDecimalState(REPLY_OTHER_CHANNEL, localDnsStatistics.getReplyOther());
+        updateDecimalState(REPLY_DNSSEC_CHANNEL, localDnsStatistics.getReplyDNSSEC());
+        updateDecimalState(REPLY_NONE_CHANNEL, localDnsStatistics.getReplyNone());
+        updateDecimalState(REPLY_BLOB_CHANNEL, localDnsStatistics.getReplyBlob());
+        updateDecimalState(DNS_QUERIES_ALL_REPLIES_CHANNEL, localDnsStatistics.getDnsQueriesAllTypes());
+        updateDecimalState(PRIVACY_LEVEL_CHANNEL, localDnsStatistics.getPrivacyLevel());
+
+        var adsPercentageToday = localDnsStatistics.getAdsPercentageToday();
+        if (adsPercentageToday != null) {
+            var state = new QuantityType<>(new BigDecimal(adsPercentageToday.toString()), PERCENT);
+            updateState(ADS_PERCENTAGE_TODAY_CHANNEL, state);
+        }
+        updateState(ENABLED_CHANNEL, OnOffType.from(localDnsStatistics.getEnabled()));
+        if (localDnsStatistics.getEnabled()) {
+            updateState(DISABLE_ENABLE_CHANNEL, new StringType(ENABLE.toString()));
+        }
+    }
+
+    private void updateDecimalState(String channelID, @Nullable Integer value) {
+        if (value == null) {
```

Comment:
Shouldn't you update to `UnDefType.NULL` in this case ?

Reply:
I don't want to. I would prefer to leave the value that was before